controller/linodemachine_controller.go

```diff
@@ -56,28 +66,33 @@ func (r *LinodeMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	ctx, cancel := context.WithTimeout(ctx, reconciler.DefaultedLoopTimeout(r.ReconcileTimeout))
 	defer cancel()
 
-	log := ctrl.LoggerFrom(ctx)
+	log := ctrl.LoggerFrom(ctx).WithName("LinodeMachineReconciler").WithValues("name", req.NamespacedName.String())
 
-	// TODO(user): your logic here
 	linodeMachine := &infrav1.LinodeMachine{}
 	if err := r.Get(ctx, req.NamespacedName, linodeMachine); err != nil {
+		log.Info("Failed to fetch Linode machine", "error", err.Error())
+
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	machine, err := util.GetOwnerMachine(ctx, r.Client, linodeMachine.ObjectMeta)
 	if err != nil {
-		return ctrl.Result{}, err
-	}
-	if machine == nil {
+		log.Info("Failed to fetch owner machine", "error", err.Error())
+
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	} else if machine == nil {
 		log.Info("Machine Controller has not yet set OwnerRef")
+
 		return ctrl.Result{}, nil
 	}
 
 	log = log.WithValues("Linode machine: ", machine.Name)
+
 	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, machine.ObjectMeta)
```

Comment:
@luthermonson @zliang-akamai It seems we trust in metadata information, but there is no guarantee to find the right cluster. I pretty understand it is an edge case, but not impossible. WDYT? Do we need some extra validation here?

Reply:
i agree with you that it's a super edge case... maybe skip extra validation for now